### ISSUE TITLE
Dynamically linked library

### ISSUE THREAD

**Issue by bkmgit** [Genivia/RE-flex#142]
Consider modifying the configure script to allow creation of either, or both statically and dynamically linked libraries.

**Comment by genivia-inc** [comment]
libtool turned out to be problematic in the past on MacOS. So I've not touched lib/Makefile.am for a long time.

**Comment by genivia-inc** [comment]
@bkmgit  It's been a while, maybe too long... Libtool isn't available on MacOS by default. Needs Homebrew or Macports to install. That's unfortunate. On the other hand, there is ranlib. Also the `build.sh` script could be modified perhaps to check if libtool is available and use it? What do you think?

**Comment by bkmgit** [comment]
Using ranlib seems like it would work.

**Comment by genivia-inc** [comment]
I don't think that ranlib alone would work, since it is part of libtool and libtool causes problems on MacOS.

I checked but MacOS still has the libtool issues. So it won't be possible on MacOS to create dynamic libraries using autotools with libtool/ranlib. Using autotools causes problems when dynamic libraries are enabled with `LT_INIT` and the `lib_LTLIBRARIES` in the lib/Makefile.am. It simply doesn't compile and I can't find a way to get around that.
